Add enchantment mutations to cached_mutations

Coolthulhu opened the pull request:
Closes #539

Note: may cause some mods to debugmsg! Only if they use stat mods or hp mods on enchantment mutations, though. Doesn't break them, they just get noisy about something that doesn't work and hasn't worked.

TODO:
* [x] Apply enchantment mutations on the first turn after load

Won't do here but will want in the future:
* Support enchantment mutations with stat mods and hp mods
* Allow mutation->enchantment->mutation->... graphs. Also starting from bionics.

olanti-p commented on src/relic.cpp:
```diff
@@ -36,12 +36,8 @@ void relic::load( const JsonObject &jo )
-                ench = ench.id.obj();
-            }
-            add_passive_effect( ench );
+            enchantment_id ench = enchantment::load_enchantment( jobj, "" );
+            add_passive_effect( ench.obj() );
```
`relic` is used in both itypes and actual items; during game start relics are first loaded by DynamicDataLoader for itypes, and then during submap/`.sav` loading - for items.

After this change, inline enchantments get their own inline ids and are added to the factory. So after finalization, each loaded item from the save that has `relic_data` with enchantments will create new entries in the factory.

Not sure how much problems that may cause (I assume factories shouldn't add elements after finalization?), but it at least breaks stacking of items with same `relic_data`.

This may probably be solved by removing relic data from items, and changing relevant code to use the item's itype's relic data instead - we don't have code that changes relic data anyway (unlike in DDA where items may have unique procgenned enchantments).

Coolthulhu replied:
> After this change, inline enchantments get their own inline ids and are added to the factory. So after finalization, each loaded item from the save that has relic_data with enchantments will create new entries in the factory.

I forgot that `relic_data` is per-instance and not per-type. This certainly complicates things.
I mostly wanted to use the factory for consistent checks everywhere, but it looks like it will be simpler to just manually check every instance where `enchantment` is being loaded.

Coolthulhu commented:
Alright, I removed the ID creation, it complicated things too much.

I added some rather aggressive checks for traits that don't work yet, but I'd say it's better to be safe than sorry.
